Device service is notified when device addressable is updated

jduranf opened the pull request:
This should fix #613.
Is implemented in the same way as it's done with profiles.

Signed-off-by: Joan Duran <jduran@circutor.com>

tsconn23 commented on internal/core/metadata/rest_addressable.go:
```diff
@@ -362,11 +362,25 @@ func restGetAddressableByAddress(w http.ResponseWriter, r *http.Request) {
 
 // Notify the associated device services for the addressable
```
@jduranf I think you'll want to update this comment to reflect your change. I see you've eliminated the notification to the Device Service itself per your comment in the issue. Just want to confirm we don't want to maintain the old behavior in addition to your change.

Once above is addressed and branch is rebased, will merge.

jduranf commented:
Yes @tsconn23, the device service should be notified for any change in it's device addressable, and there is no point to notify the device service for any change in it's addressable.

Fixed the comments and rebased
